includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -917,7 +941,11 @@ protected function prepare_payment_information( $order ) {
 
 		if ( ! empty( $_POST[ 'wc-' . static::GATEWAY_ID . '-new-payment-method' ] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
 			// During normal orders the payment method is saved when the customer enters a new one and chooses to save it.
-			$payment_information->must_save_payment_method();
+			$payment_information->must_save_payment_method_to_store();
+		}
+
+		if ( ! empty( $_POST['save_user_in_platform_checkout'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
```

Comment:
Good call. Fixed in abd176b.

Reply:
Thanks for looking into this. 

My only concern now is if `$_POST['save_user_in_platform_checkout']` is equal to the boolean value `true`. Then `'true' === $_POST['save_user_in_platform_checkout']` would resolve to `false` since `'true' !== true`. 

Could we do something like the following:
```php
...
// phpcs:ignore WordPress.Security.NonceVerification.Missing
$should_save_user_in_platform_checkout = 
    ! empty( $_POST['save_user_in_platform_checkout'] )
    && true === filter_var( $_POST['save_user_in_platform_checkout'], FILTER_VALIDATE_BOOLEAN ) ;
if ( $should_save_user_in_platform_checkout ) {
    $payment_information->must_save_payment_method_to_platform();
}
...
```

Source: [How to convert string to boolean PHP](https://stackoverflow.com/a/15075609).

Reply:
I agree with the suggestion, totally forgot about `filter_var`. I guess we don't need `===` because `filter_var` will always return according to the second argument. Fixed in 43dfbdc.

Reply:
Good call 🙂